### PR TITLE
ENG-8258: Slack client upgrade

### PR DESCRIPTION
**Pull request by waltfy**
https://linear.app/incident-io/issue/ENG-8258/2-slack-client-upgrade

---

As part of the Slack Enterprise Grid project, we need to upgrade the Slack client to the latest version.

This pull-request takes care of that by pulling all the upstream changes and ensuring that we apply our patches over it.

---

TODO:
- [ ] Document process for upgrading
- [ ] Document process for releasing

### CODE REVIEW

File: block_element.go [hunk 1]
```diff
@@ -171,12 +171,6 @@ func (s *ButtonBlockElement) WithStyle(style Style) *ButtonBlockElement {
 	return s
 }
 
-// WithConfirm adds a confirmation dialogue to the button object and returns the modified ButtonBlockElement
-func (s *ButtonBlockElement) WithConfirm(confirm *ConfirmationBlockObject) *ButtonBlockElement {
```
Comment (isaacseymour) [inline block_element.go]: we're losing this - sure we don't want it?

File: info.go [hunk 2]
```diff
@@ -406,6 +410,9 @@ func (t *JSONTime) UnmarshalJSON(buf []byte) error {
 	s := bytes.Trim(buf, `"`)
 
 	if string(s) == "null" {
+		// Question: the line below is what upstream
```
Comment (isaacseymour) [inline info.go:413]: Have you figured this out? fwiw I don't think we use this method, so we should just use the upstream

**Review by isaacseymour — APPROVED** [review]
seems legit

File: block_element.go (hunk 1 above)
Comment (waltfy) [reply block_element.go]: We're adding this, no? If you look at [`e56f38b` (#12)](https://github.com/incident-io/slack/pull/12/commits/e56f38bc7dacfdc521d59d842325198983524612#diff-53b57cd5be231077bacd61bb89ad77ef9bdd61ae4a6eced4a785f434bf508e16R174-R179)

File: info.go (hunk 2 above)
Comment (waltfy) [reply info.go:413]: Ok, nice - that's the context I was after - will fix!

**Pull request merged**